Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -459,36 +459,56 @@ void AppendPipeline::didReceiveInitializationSegment()
             trackIndex++;
         }
     } else {
-        // Since we don't rely on the demuxer pad-added signal and this pipeline is not
-        // stream-aware, we need to account for stream topology changes ourselves.
-        unsigned videoPadsCount = 0;
-        unsigned audioPadsCount = 0;
-        unsigned textPadsCount = 0;
+        HashSet<String> videoPadStreamIDs;
+        HashSet<String> audioPadStreamIDs;
+        HashSet<String> textPadStreamIDs;
         for (auto pad : GstIteratorAdaptor<GstPad>(GUniquePtr<GstIterator>(gst_element_iterate_src_pads(m_demux.get())))) {
-            if (gst_pad_is_linked(pad))
-                continue;
             auto [parsedCaps, streamType, presentationSize] = parseDemuxerSrcPadCaps(adoptGRef(gst_pad_get_current_caps(pad)).get());
+            UNUSED_VARIABLE(parsedCaps);
+            UNUSED_VARIABLE(presentationSize);
+            GUniquePtr<char> streamIDAsCharacters(gst_pad_get_stream_id(pad));
+            String streamID = String::fromLatin1(streamIDAsCharacters.get());
             if (streamType == StreamType::Audio)
-                audioPadsCount++;
+                audioPadStreamIDs.add(WTFMove(streamID));
             else if (streamType == StreamType::Video)
-                videoPadsCount++;
+                videoPadStreamIDs.add(WTFMove(streamID));
             else if (streamType == StreamType::Text)
-                textPadsCount++;
+                textPadStreamIDs.add(WTFMove(streamID));
         }
 
         unsigned videoTracksCount = 0;
         unsigned audioTracksCount = 0;
         unsigned textTracksCount = 0;
+        bool doAudioTrackStreamIDsMatch = true;
+        bool doVideoTrackStreamIDsMatch = true;
+        bool doTextTrackStreamIDsMatch = true;
         for (const auto& track : m_tracks) {
-            if (track->streamType == StreamType::Audio)
+            GUniquePtr<char> streamIDAsCharacters(gst_pad_get_stream_id(track->entryPad.get()));
+            String streamID = String::fromLatin1(streamIDAsCharacters.get());
+            if (track->streamType == StreamType::Audio) {
                 audioTracksCount++;
-            else if (track->streamType == StreamType::Video)
+                if (streamID.isEmpty() || !audioPadStreamIDs.contains(streamID))
```

Comment:
Me too.

Reply:
TL&DR

If I added it is because it was in layout test that crashed because of that but I lost track of all regressions I fixed while iterating on this patch.